From f3bc766dbcc0dcbd078bf781e042972ac4aafa47 Mon Sep 17 00:00:00 2001
From: Liu Linhong <linhong.liu@intel.com>
Date: Fri, 10 Apr 2015 17:00:26 +0800
Subject: [PATCH] Set cdev's otg descriptor in android_bind_config

Set cdev's otg descriptor in android_bind is not working. This is
Because we set is_otg = 1 in phy-intel-cht.c. But the cht otg
initialization code is not called before android_bind. So set
cdev's otg descriptor in android_bind_config to resolve this.

Change-Id: Ibd217fd840224668574ec749f269355158a28695
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-9620
Signed-off-by: Liu Linhong <linhong.liu@intel.com>
---
 drivers/usb/gadget/android.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index a3a4d7c..845207d 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -1487,8 +1487,10 @@ static int android_bind_config(struct usb_configuration *c)
 	struct android_dev *dev = _android_dev;
 	int ret = 0;
 
-	if (gadget_is_otg(c->cdev->gadget))
+	if (gadget_is_otg(c->cdev->gadget)) {
 		c->descriptors = otg_desc;
+		c->cdev->otg_desc = &otg_descriptor;
+	}
 
 	ret = android_bind_enabled_functions(dev, c);
 	if (ret)
@@ -1550,9 +1552,6 @@ static int android_bind(struct usb_composite_dev *cdev)
 	device_desc.iSerialNumber = id;
 	dev->reset_string_id = id;
 
-	if (gadget_is_otg(gadget))
-		cdev->otg_desc = &otg_descriptor;
-
 	usb_gadget_set_selfpowered(gadget);
 	dev->cdev = cdev;
 
-- 
1.9.1

